internal/npm_install/generate_build_file.js

```diff
@@ -317,15 +344,57 @@ function flattenDependencies(pkg, dep, pkgsMap) {
  */
 function printJson(pkg) {
   // Clone and modify _dependencies to avoid circular issues when JSONifying
+  // & delete _files array
   const cloned = {...pkg};
   cloned._dependencies = cloned._dependencies.map(dep => dep._dir);
+  delete cloned._files;
   return JSON.stringify(cloned, null, 2).split('\n').map(line => `# ${line}`).join('\n');
 }
 
+/**
+ * A filter function for a bazel filegroup.
+ * @param files array of files to filter
+ * @param exts list of white listed extensions; if empty, no filter is done on extensions;
+ *             '' empty string denotes to allow files with no extensions, other extensions
+ *             are listed with '.ext' notation such as '.d.ts'.
+ */
+function filterFilesForFilegroup(files, exts = []) {
+  files = files
+              // Bazel target names must be composed entirely of characters drawn
+              // from the set a–z, A–Z, 0–9, and the punctuation symbols _/.+-=,@~.$
+              // Files with spaces are also not allowed in Bazel runfiles
+              // See https://github.com/bazelbuild/bazel/issues/4327
+              .filter(f => !f.match(/[^a-zA-Z0-9\_\/\.\+\-\=\,\@\~\$]/))
+              // Don't include bazel BUILD and WORKSPACE files in a filegroup
+              .filter(f => path.basename(f) !== 'BUILD')
+              .filter(f => path.basename(f) !== 'BUILD.bazel')
+              .filter(f => path.basename(f) !== 'WORKSPACE');
+  if (exts.length) {
+    const allowNoExts = exts.includes('');
+    files = files.filter(f => {
+      // include files with no extensions if noExt is true
+      if (allowNoExts && !path.extname(f)) return true;
+      // filter files in exts
+      let found = false;
+      exts.forEach(e => {
```

Comment:
nit: if you use a for loop instead of forEach, then you could just return instead of having the `found` variable

Reply:
👍